### PR TITLE
Support multiple variables per single ENV cmd

### PR DESCRIPTION
**Pull request by matez**
Add `environmentVariable(Map<String, String>)` and `environmentVariable(Closure)` methods to Dockerfile task. These two will generate ENV command in form

`ENV "foo"="bar" "baz"="bah"`

Original method `environmentVariable(String, String)` keep generating

`ENV foo bar`

### CODE REVIEW

**Comment by cdancy** [comment]
@matez thanks for the PR! Any thoughts on combining `EnvironmentVariableInstruction` and `MultiEnvironmentVariableInstruction` into 1 class? I'm thinking we keep the name `EnvironmentVariableInstruction` but replace its contents with your Multi impl. Then massage the methods that use `EnvironmentVariableInstruction` the old way to work with your impl. Hope that makes sense.

**Comment by matez** [comment]
I've added `MultiEnvironmentVariableInstruction` just to be able to extend `MapInstruction`. I'll update it to have only `EnvironmentVariableInstruction` supporting both approaches.

**Comment by matez** [comment]
I can do

```groovy
static class EnvironmentVariableInstruction extends MapInstruction {
        EnvironmentVariableInstruction(String key, String value) {
            super([(key): value])
        }

        EnvironmentVariableInstruction(Map envVars) {
            super(envVars)
        }

        EnvironmentVariableInstruction(Closure envVars) {
            super(envVars)
        }

        @Override
        String getKeyword() {
            "ENV"
        }
    }
```

but it'll change old behavior for `environmentVariable(String, String)` from `ENV foo bar` to `ENV "foo"="bar"`. I am not sure if such change is breaking or not ... since dunno when docker introduced ENV  cmd with multiple varibales

**Comment by matez** [comment]
Since 1.4.0 https://github.com/docker/docker/blob/master/CHANGELOG.md#140-2014-12-11

**Comment by cdancy** [comment]
@matez I think that should be Ok as we don't support docker versions that old anyway. So long as all integration tests pass we should be good.

@orzeh any thoughts one way or the other?

**Comment by matez** [comment]
81a19b5 shouldn't change any previous behavior ... There is only `EnvironmentVariableInstruction` used in all `environmentVariable` methods.

**Comment by cdancy** [comment]
@matez honestly I had a completely different way of going about doing this in my head but I prefer you're take on things. LGTM! 

@orzeh second pair of eyes for a code review?

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy [hunk 1]
```diff
@@ -18,6 +18,7 @@ package com.bmuschko.gradle.docker.tasks.image
 import org.gradle.api.DefaultTask
 import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
+import org.gradle.internal.impldep.com.google.common.collect.Maps
```
Comment (cdancy) [inline src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy]: I don't see this import being used anywhere. Further more I think we would want to shy away from using anything internal to gradles API.

Comment (matez) [inline src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy]: Totally agree ... missed that :(

**Comment by orzeh** [comment]
I don't see any issues. @matez thanks for this PR!

@cdancy please merge.

**Comment by cdancy** [comment]
All integrations tests pass as expected (couple failures but un-related and will cleanup in a separate PR). Thanks @matez !

**Pull request merged**